lib/Backend/NativeCodeGenerator.cpp

```diff
@@ -2802,25 +2810,40 @@ void
 NativeCodeGenerator::ProfilePrint()
 {
     Js::ScriptContextProfiler *codegenProfiler = this->backgroundCodeGenProfiler;
+
     if (Js::Configuration::Global.flags.Verbose)
     {
         //Print individual CodegenProfiler information in verbose mode
         while (codegenProfiler)
         {
-            codegenProfiler->ProfilePrint(Js::Configuration::Global.flags.Profile.GetFirstPhase());
+            if (codegenProfiler->IsInitialized())
+            {
+                codegenProfiler->ProfilePrint(Js::Configuration::Global.flags.Profile.GetFirstPhase());
+            }
             codegenProfiler = codegenProfiler->next;
         }
     }
     else
     {
         //Merge all the codegenProfiler for single snapshot.
-        codegenProfiler = codegenProfiler->next;
-        while (codegenProfiler)
+        Js::ScriptContextProfiler* mergeToProfiler = codegenProfiler;
+        while (mergeToProfiler != nullptr && !mergeToProfiler->IsInitialized())
```

Comment:
Below loop doesn't ensure the first profiler (mergeToProfiler) is initialized.

---
In reply to: [66282264](https://github.com/Microsoft/ChakraCore/pull/1096#discussion_r66282264) [](ancestors = 66282264)